<?xml version="1.0"?>

<!-- rover_gazebo_control.launch

Use Gazebo to simulate a rover with Ackermann steering.

Arguments:
  namespace (string, default: rover)
    rover namespace.
  world_name (string, default: worlds/empty.world)
      Gazebo world name.
  cmd_timeout (double, default: 0.5)
      Command timeout passed to the ackermann_controller node.
  x (double, default: 0.0)
  y (double, default: 0.0)
  z (double, default: 0.1)
  roll (double, default: 0.0)
  pitch (double, default: 0.0)
  yaw (double, default: 0.0)
      Vehicle pose. x, y, and z are measured in meters. roll, pitch, and yaw
      are measured in radians.
-->

<launch>
    <!-- Namespace -->
    <arg name="namespace" default="rover"/>
    <arg name="start_plt" default="true"/>

    <!-- rover initial pose -->
    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="0.0"/>
    <arg name="roll" default="0.0"/>
    <arg name="pitch" default="0.0"/>
    <arg name="yaw" default="0.0"/>

    <!-- Launch Rover -->
    <include file="$(find rover_desc_pkg)/launch/rover.launch">
        <arg name="namespace" value="$(arg namespace)"/>
    </include>

    <!--  &lt;!&ndash; Launch Plot Juggler &ndash;&gt;-->
    <!--  <node name="plotjuggler" pkg="plotjuggler" type="PlotJuggler" args="&#45;&#45;nosplash"-->
    <!--        if="$(eval start_plt)"/>-->

    <group ns="$(arg namespace)">
        Launch Gazebo World
        <include file="$(find rover_gazebo_control)/launch/rover_gazebo_world.launch"/>

        <!-- Spawn the rover. -->
        <node name="rover_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
              args="-urdf -param robot_description -model rover
                -gazebo_namespace /$(arg namespace)/gazebo
                -x $(arg x) -y $(arg y) -z $(arg z)
                -R $(arg roll) -P $(arg pitch) -Y $(arg yaw)"/>

        <node name="rover_ctrlr_spawner" pkg="controller_manager" type="spawner"
              respawn="false" output="screen" ns="rover"
              args="rover_joint_state_ctrlr
                      front_right_steering_ctrlr
                      front_left_steering_ctrlr
                      front_right_wheel_ctrlr
                      front_left_wheel_ctrlr
                      rear_right_wheel_ctrlr
                      rear_left_wheel_ctrlr"/>
        <!--        Load the joint controllers. One of these publishes the joint states to joint_states.-->
        <!--        <node name="rover_ctrlr_spawner" pkg="controller_manager" type="spawner"-->
        <!--              args="$(find rover_gazebo_control)/config/rover_joint_ctrlr_params.yaml"/>-->
    </group>
</launch>
